File: RepoTransBench/source_projects/Python/tomasbasham_ratelimit/public_tests/test_public_exception_test.py
from ratelimit.exception import RateLimitException

def test_public_rate_limit_exception_inheritance():
    ex = RateLimitException("foo", 0.99)
    assert isinstance(ex, Exception)
    assert hasattr(ex, 'period_remaining')

def test_public_exception_str_and_value():
    ex = RateLimitException("overload", 2)
    assert str(ex) == "overload"
    assert ex.period_remaining == 2

File: RepoTransBench/target_projects/Python/C++/tomasbasham_ratelimit/public_tests/test_public_exception_test.cpp
#include <gtest/gtest.h>
#include "ratelimit/exception.h"

TEST(PublicExceptionTest, RateLimitExceptionInheritance) {
    ratelimit::RateLimitException ex("foo", 0.99);
    EXPECT_TRUE(dynamic_cast<std::exception*>(&ex) != nullptr);
    EXPECT_NO_THROW(ex.period_remaining);
}

TEST(PublicExceptionTest, ExceptionStrAndValue) {
    ratelimit::RateLimitException ex("overload", 2);
    EXPECT_STREQ(ex.what(), "overload");
    EXPECT_EQ(ex.period_remaining, 2);
}